pkg/kubelet/apis/config/validation/validation_linux.go

```diff
@@ -28,9 +28,14 @@ import (
 
 // validateKubeletOSConfiguration validates os specific kubelet configuration and returns an error if it is invalid.
 func validateKubeletOSConfiguration(kc *kubeletconfig.KubeletConfiguration) error {
-	if kc.FailCgroupV1 && !libcontainercgroups.IsCgroup2UnifiedMode() {
+	isCgroup1 := !libcontainercgroups.IsCgroup2UnifiedMode()
+	if kc.FailCgroupV1 && isCgroup1 {
 		return fmt.Errorf("kubelet is configured to not run on a host using cgroup v1. cgroup v1 support is in maintenance mode")
 	}
 
+	if isCgroup1 && kc.SingleProcessOOMKill != nil && !*kc.SingleProcessOOMKill {
+		return fmt.Errorf("invalid configuration: singleProcessOOMKill must be specified nil and implicitly converted to true when using cgroup v1")
```

Comment:
This error message seems to suggest that `singleProcessOOMKill` can only be `nil` for cgroup v1 linux. However the API documentation says that both `nil` and explicit `true` are allowed. Shall we update this error message to make it clear that only explicit `false` is disallowed (i.e. both `nil` and explicit `true` are allowed)?

Reply:
Done.
https://github.com/kubernetes/kubernetes/pull/126096/files#diff-d2cf199db99b0d2513c403aa1ae8c56e5e7c9b98345189bb8af50be0c3d7fd9aR37